Expose kube-api burst/qps settings for kube components

spiffxp opened the pull request:
Trying to remove API limits to assist with tracking down the cause of #14216

Default to 20.0 qps, 30 burst since that's what I saw hardcoded in at
least some of the components

Unclear if maybe it'd be better to just assume these are set as part of
the incoming kubeconfig.  For now just exposing them as flags since it's
easier for me to manually tweak.

spiffxp commented:
I think #14955 will obviate this, and I'm not a big fan of adding yet more flags, but it's been useful for testing purposes

dchen1107 commented on cmd/kube-controller-manager/app/controllermanager.go:
```diff
@@ -189,6 +193,8 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.Kubeconfig, "kubeconfig", s.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
 	fs.StringVar(&s.RootCAFile, "root-ca-file", s.RootCAFile, "If set, this root certificate authority will be included in service account's token secret. This must be a valid PEM-encoded CA bundle.")
 	fs.BoolVar(&s.EnableExperimental, "enable-experimental", s.EnableExperimental, "Enables experimental controllers (requires enabling experimental API on apiserver).")
+	fs.Float32Var(&s.KubeApiQps, "kube-api-qps", s.KubeApiQps, "QPS to use while talking with kubernetes api")
```
for consistency, s/api/apiserver in flag description.

dchen1107 commented on cmd/kube-controller-manager/app/controllermanager.go:
```diff
@@ -189,6 +193,8 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.RootCAFile, "root-ca-file", s.RootCAFile, "If set, this root certificate authority will be included in service account's token secret. This must be a valid PEM-encoded CA bundle.")
 	fs.BoolVar(&s.EnableExperimental, "enable-experimental", s.EnableExperimental, "Enables experimental controllers (requires enabling experimental API on apiserver).")
+	fs.Float32Var(&s.KubeApiQps, "kube-api-qps", s.KubeApiQps, "QPS to use while talking with kubernetes api")
+	fs.IntVar(&s.KubeApiBurst, "kube-api-burst", s.KubeApiBurst, "Burst to use while talking with kubernetes api")
```
ditto.

dchen1107 commented on cmd/kubelet/app/server.go:
```diff
@@ -201,6 +204,8 @@ func NewKubeletServer() *KubeletServer {
 		RootDirectory:     defaultRootDir,
 		SyncFrequency:     10 * time.Second,
 		SystemContainer:   "",
+		KubeApiQps:        20.0,
```
I am ok with current default setting. But if I remembered correctly, kubelet's default value is 5, not 20.

dchen1107 commented:
LGTM except some nits.

spiffxp commented:
Addressed nits, rebased to avoid squashing #15215

timothysc commented:
fwiw these were exposed many moons ago then someone hid them on the run-up to 1.0. 

ಠ_ಠ

lgtm

dchen1107 commented:
LGTM

spiffxp commented:
rebase flail

spiffxp commented:
another dummy change to see if e2e tests will show up this time

spiffxp committed:
Expose kube-api burst/qps settings for kube components

Default to hardcodes for components that had them, and 5.0 qps, 10 burst
for those that relied on client defaults

Unclear if maybe it'd be better to just assume these are set as part of
the incoming kubeconfig.  For now just exposing them as flags since it's
easier for me to manually tweak.

dchen1107 commented:
GCE e2e failure is caused by known flaky test: deployment should scale up and down in the right order 

ok to test
